client/header/activity-panel/unread-indicators.js

```diff
@@ -8,6 +8,9 @@ import { DEFAULT_ACTIONABLE_STATUSES } from 'wc-api/constants';
 export function getUnreadNotes( select ) {
 	const { getCurrentUserData, getNotes, getNotesError, isGetNotesRequesting } = select( 'wc-api' );
 	const userData = getCurrentUserData();
+	if ( ! userData ) {
```

Comment:
Is this needed? `getCurrentUserData` already [has a fallback](https://github.com/woocommerce/woocommerce-admin/blob/master/client/wc-api/user/selectors.js#L12).

Reply:
It doesn't have a fallback in the current iteration as it's using the global directly:

```js
const getCurrentUserData = ( getResource, requireResource ) => (
	requirement = DEFAULT_REQUIREMENT
) => {
	const initialState = wcSettings.currentUserData;
	return requireResource( requirement, 'current-user-data' ).data || initialState;
};
```

After refactor in future iterations it will have a fallback.  The existing logic is fragile and will break things if for some reason currentUser isn't present (which is entirely possible if the global is clobbered).

Reply:
Good point. In any case, I think it would make sense to handle the missing data case inside `getCurrentUserData()` instead of having to check if `userData` is defined in every consumer.

Reply:
> In any case, I think it would make sense to handle the missing data case inside getCurrentUserData() instead of having to check if userData is defined in every consumer.

I agree, and that will be the case once it's refactored (and I'll remove the instance of checking for userData here at that point).

Reply:
https://github.com/woocommerce/woocommerce-admin/issues/2943